mission/functions/tasks/primary/fn_task_pri_prepare.sqf

```diff
@@ -105,13 +105,27 @@ _taskDataStore setVariable ["fnc_subtaskGoAway", {
 
 	[_taskStore, "ColorBlack"] call (_taskStore getVariable "fnc_changeAreaMarkerColor");
 
+	// delete generated sites as early as possible
+	// doesn't matter if players are inside the zone
+	// make it clear they shouldn't be here by removing things in front of them
+	// GTFO now please!
+	if (scriptDone (_taskStore getVariable "generateHandle")) then {
+		call vn_mf_fnc_sites_delete_all_active_sites;
+		call vn_mf_fnc_daccong_respawns_delete_all;
+	};
+
 	/*
 	players have left the AO's blue circle
 	we're good to end this task and move back to the DC prepare task
+
+	NOTE -- make sure to check for scriptDone here too.
 	*/
-	if ((count _playersInArea) == 0) exitWith {
+
+	if ((count _playersInArea) == 0 && (scriptDone (_taskStore getVariable "generateHandle"))) exitWith {
```

Comment:
merge this with the `if (scriptDone (_taskStore getVariable "generateHandle")) exitWith {` block below?

do a nested `if` on the player count check?

Reply:
scratch that, not for the same thing